src/etcd/model/base.cr

```diff
@@ -7,6 +7,18 @@ module Etcd::Model
     include JSON::Serializable
   end
 
+  abstract struct WithHeader < Base
+    getter header : Header
+  end
+
+  struct Error < Base
+    getter details : Array(NamedTuple(type_url: String, value: String))
```

Comment:
I think use a struct for the details ------ that way you can use a method accessor